lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/util/XmlReader.java

```diff
@@ -51,31 +51,42 @@ public static boolean hasServerRoot(Path filePath) {
     }
 
     public static boolean hasServerRoot(File xmlFile) {
-        XMLEventReader reader = null;
+        if (!xmlFile.exists() || xmlFile.length() == 0) {
+            return false;
+        }
         
         try {
-            if (!xmlFile.exists() || xmlFile.length() == 0) {
-                return false;
-            }
-
             XMLInputFactory factory = XMLInputFactory.newInstance();
-            reader = factory.createXMLEventReader(new FileInputStream(xmlFile));
-            if (reader.hasNext()) {
-                XMLEvent firstTag = reader.nextTag(); // first start/end element
-                reader.close();
-                return isServerElement(firstTag);
+            try {
+                factory.setProperty(XMLInputFactory.IS_VALIDATING, Boolean.FALSE);
+                factory.setProperty(XMLInputFactory.IS_SUPPORTING_EXTERNAL_ENTITIES, Boolean.FALSE);
+                factory.setProperty(XMLInputFactory.SUPPORT_DTD, Boolean.FALSE);
+            } catch (Exception e) {
+                LOGGER.warning("Could not set properties on XMLInputFactory.");
             }
-        } catch (FileNotFoundException e) {
-            LOGGER.severe("Unable to access file "+ xmlFile.getAbsolutePath());
-        } catch (XMLStreamException e) {
-            LOGGER.severe("Error received trying to read XML file: " + xmlFile.getAbsolutePath());
-        } finally {
-            if (reader != null) {
-                try {
-                    reader.close();
-                } catch (Exception ignored) {   
+
+            XMLEventReader reader = null;
+
+            try (FileInputStream fis = new FileInputStream(xmlFile)) {
+                reader = factory.createXMLEventReader(fis);
+                while (reader.hasNext()) {
+                    XMLEvent nextEvent = reader.nextEvent();
+                    if (nextEvent.isStartElement()) {
+                        return isServerElement(nextEvent);
+                    }
```

Comment:
I know we never accounted for this before, but does anyone ever add an [XML prolog](https://www.w3schools.com/XML/xml_syntax.asp) to a server config xml?
Wondering if this would fail it, even if there is a `<server>` root tag after the XML prolog.

ie. 
```
<?xml version="1.0" encoding="UTF-8"?> 
<server>
...
</server>
```

Reply:
I also noticed the switch from `nextTag()` to `nextEvent()`. It seems `nextTag()` will skip some insignificant events until a start/end element is found.

Reply:
It handles this because it only looks at the first start element, and that is not a start element. It is special. I can add one to an existing test case to prove it.

Reply:
Actually, we already have `src/test/resources/sample/custom_server.xml` that contains that XML prolog. It is used in `XmlReaderTest` and passes both the `hasServerRoot` check and `isConfigXMLFile` check.